### PR TITLE
Add error message to compile when Node is missing

### PR DESCRIPTION
**Pull request by lrbison**
*Description of changes:*

Add error message to compile when Node is missing

---
The message:

```
   Compiling aperf v0.1.0 (/fsx/lrbison/code/aperf)
error: failed to run custom build command for `aperf v0.1.0 (/fsx/lrbison/code/aperf)`

Caused by:
  process didn't exit successfully: `/fsx/lrbison/code/aperf/target/debug/build/aperf-d4acbb2dac42850a/build-script-build` (exit status: 1)
  --- stdout
  cargo:rustc-env=VERGEN_GIT_SHA=c88030f
  cargo:rerun-if-changed=.git/HEAD
  cargo:rerun-if-changed=.git/refs/heads/build
  cargo:rerun-if-changed=build.rs
  cargo:rerun-if-env-changed=VERGEN_IDEMPOTENT
  cargo:rerun-if-env-changed=SOURCE_DATE_EPOCH
  cargo:rerun-if-changed=package.json
  cargo:rerun-if-changed=package-lock.json
  Build requires npm, but it was not found.  Please install Node >= 16.16.0
```

---
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: build.rs [hunk 1]
```diff
@@ -7,10 +7,18 @@ fn main() -> Result<()> {
 
     println!("cargo:rerun-if-changed=package.json");
     println!("cargo:rerun-if-changed=package-lock.json");
-    let status = Command::new("npm").arg("install").spawn()?.wait()?;
-    if !status.success() {
+    let proc = Command::new("npm").arg("install").spawn();
+    if let Err(_proc) = proc {
+        println!("Build requires npm, but it was not found.  Please install Node >= 16.16.0");
```
Comment (wash-amzn) [inline build.rs:12]: 2 spaces between the sentences, and the second sentence not period-ed?

File: build.rs [hunk 2]
```diff
@@ -7,10 +7,18 @@ fn main() -> Result<()> {
...
         std::process::exit(1);
     }
+    if let Ok(mut child) = proc {
+        let status = child.wait()?;
+        if !status.success() {
+            std::process::exit(1);
+        }
+    }
```
Comment (wash-amzn) [inline build.rs:20]: This should be doable like
```
match Command::new("npm").arg("install").spawn() {
  Err(_proc) => {
    // stuff
  },
  Ok(child) => {
    // stuff
  }
}
```

File: build.rs [hunk 3]
```diff
@@ -7,10 +7,19 @@ fn main() -> Result<()> {
...
+    match Command::new("npm").arg("install").spawn() {
+        Err(_proc) => {
+            println!("Build requires npm, but it was not found. Please install Node >= 16.16.0.");
+            std::process::exit(1);
+        }
+        Ok(mut child) => {
+            let status = child.wait()?;
+            if !status.success() {
+                std::process::exit(1);
+            }
```
Comment (wash-amzn) [inline build.rs:19]: What does it look like if this case is it? The process' stderr is still printed, right? I just want to make sure we don't have a mysterious termination without being able to figure out what went wrong.

Still might want to have aperf print a line stating that the command exited unsuccessfully (and provide the command that failed).

Comment (lrbison) [reply build.rs]: Added

File: build.rs [hunk 4]
```diff
@@ -7,10 +7,20 @@ fn main() -> Result<()> {
 
     println!("cargo:rerun-if-changed=package.json");
     println!("cargo:rerun-if-changed=package-lock.json");
-    let status = Command::new("npm").arg("install").spawn()?.wait()?;
-    if !status.success() {
-        std::process::exit(1);
+    match Command::new("npm").arg("install").spawn() {
+        Err(_proc) => {
```
Comment (janaknat) [inline build.rs:11]: `Err(_)` should also work.

**Review by wash-amzn — APPROVED** [review]

**Pull request merged**